src/main/java/hudson/remoting/Pipe.java

```diff
@@ -97,6 +98,7 @@
  *
  * @author Kohsuke Kawaguchi
  */
+@SuppressFBWarnings(value = "DESERIALIZATION_GADGET", justification = "Serializable only over remoting.")
```

Comment:
Good to mention the serialization hygiene again (JEP-200) as it's not immediately obvious what is being suppressed here (various `ObjectInputStream` usage below). Can this be narrowed to just the `readObject` method?

Reply:
It looks like the annotation can be narrowed to just the `readObject` method. It reports the finding on the entire class so that's where I put the suppression, but putting it on the method also quiets it.